src/components/views/context_menus/MessageContextMenu.tsx

```diff
@@ -236,30 +322,51 @@ export default class MessageContextMenu extends React.Component<IProps, IState>
         });
     }
 
-    private getPendingReactions(): MatrixEvent[] {
-        return this.getReactions(e => canCancel(e.status));
+    private getUnsentReactions(): Array<MatrixEvent> {
+        return this.getReactions(e => e.status === EventStatus.NOT_SENT);
     }
 
-    private getUnsentReactions(): MatrixEvent[] {
-        return this.getReactions(e => e.status === EventStatus.NOT_SENT);
+    private getSelectedText(): string {
+        return window.getSelection().toString();
+    }
+
+    private getPermalink(): string {
+        if (!this.props.permalinkCreator) return null;
+        // XXX: if we use room ID, we should also include a server where the event can be found (other than in the domain of the event ID)
```

Comment:
in theory if the permalinkCreator has context of the room then it should be able to pick servers for you. At least that's how I remember it working...

Reply:
That seems to be the case, this comment was copied from somewhere, though I don't remember where that was